sdk/src/main/java/tech/relaycorp/sdk/Message.kt

```diff
@@ -0,0 +1,57 @@
+package tech.relaycorp.sdk
+
+import java.time.ZonedDateTime
+import java.util.*
+
+abstract class Message(
+    val id: Message.Id,
+    val message: ByteArray,
+    senderEndpoint: Endpoint,
+    receiverEndpoint: Endpoint,
+    val creationDate: ZonedDateTime = ZonedDateTime.now(),
+    val expirationDate: ZonedDateTime = maxExpirationDate()
+) {
+    companion object {
+        internal fun maxExpirationDate() = ZonedDateTime.now().plusDays(30)
+    }
+
+    class Id
+    private constructor(
+        val value: String
+    ) {
+        companion object {
+            fun generate() = Id(UUID.randomUUID().toString())
+        }
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (other !is Id) return false
+            if (value != other.value) return false
+            return true
+        }
+
+        override fun hashCode() = value.hashCode()
+    }
+}
+
+class IncomingMessage internal constructor(
```

Comment:
Reminder: this needs an `ack()` method (or field, if using the constructor param as is).

Reply:
Right, I realised it when I moved the code from echo to the SDK.